lib/index.js

```diff
@@ -18,9 +18,28 @@ KeenLibrary.prototype.query = function(a, b = undefined, options = {}){
   let analysisType = a;
   let queryParams = b;
 
+  const isAnalysisTypeAnObject = typeof analysisType === 'object';
+  const savedRegex = /saved/;
+  const isSavedAnalysisType = (isAnalysisTypeAnObject && savedRegex.test(analysisType.analysis_type))
+    || savedRegex.test(analysisType);
+  
+    // validate params
+  if (!isSavedAnalysisType) {
+    if (!this.config.masterKey || !this.config.readKey) throw new Error('Please provide valid API key');
+    if ((isAnalysisTypeAnObject && !analysisType.timeframe) || (queryParams && !queryParams.timeframe)) {
+      throw new Error('Queries must contain a time range. Please provide timeframe parameter');
+    }
+    if ((isAnalysisTypeAnObject && !analysisType.event_collection) || (queryParams && !queryParams.event_collection)) {
+      throw new Error('Please provide event collection parameter');
+    }
+    if ((isAnalysisTypeAnObject && !analysisType.analysis_type) || !analysisType) {
+      throw new Error('Please provide analysis type');
+    }
+  }
+
```

Comment:
move all of this code under line :54 - this way you'll be able to use `analysisType` and `queryParams`, without `isAnalysisTypeAnObject`